src/filter.rs

```diff
@@ -2,6 +2,185 @@ use core::convert::TryInto;
 
 use crate::{common::BytesPerPixel, Compression};
 
+#[cfg(feature = "unstable")]
+mod simd {
+    // Optional module that contains portable_simd versions of the most important unfiltering algorithms.
+    use core::convert::TryInto;
+    use core::simd::cmp::{SimdOrd, SimdPartialOrd};
+    use core::simd::num::{SimdInt, SimdUint};
+    use core::simd::{LaneCount, Simd, SupportedLaneCount};
+
+    // Import the fastest arch-specific scalar implementations from the outer crate.
+    #[cfg(target_arch = "aarch64")]
+    use crate::filter::filter_paeth as filter_paeth_decode;
+    #[cfg(target_arch = "x86_64")]
+    use crate::filter::filter_paeth_stbi as filter_paeth_decode;
+
+    // AArch64 Paeth predictor. Ported from the libpng implementation at
+    // https://github.com/pnggroup/libpng/blob/master/arm/filter_neon_intrinsics.c
```

Comment:
It is better to use [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html#doc-comments) (starting with `///`) to document methods since they are treated specially by Rust tooling.